build/source/engine/coupled_em.f90

```diff
@@ -219,38 +219,54 @@ lightSnow            ! maximum interception capacity an inverse function of new
  ! compute the total number of snow and soil layers
  nLayers = nSnow + nSoil
 
- scalarCanopyLiq => mvar_data%var(iLookMVAR%scalarCanopyLiq)%dat(1)
- scalarCanopyIce => mvar_data%var(iLookMVAR%scalarCanopyIce)%dat(1)
- balanceCanopyWater0 = scalarCanopyLiq + scalarCanopyIce
-
- ! point to model state variables
- ! NOTE: need to do this at the start of each sub-step because number of layers may change
- mLayerDepth          => mvar_data%var(iLookMVAR%mLayerDepth)%dat(nSnow+1:nLayers)           ! depth of each soil layer (m)
- mLayerVolFracIce     => mvar_data%var(iLookMVAR%mLayerVolFracIce)%dat(nSnow+1:nLayers)      ! volumetric ice content in each soil layer (-)
- mLayerVolFracLiq     => mvar_data%var(iLookMVAR%mLayerVolFracLiq)%dat(nSnow+1:nLayers)      ! volumetric liquid water content in each soil layer (-)
- scalarAquiferStorage => mvar_data%var(iLookMVAR%scalarAquiferStorage)%dat(1)                ! aquifer storage (m)
+ ! allocate space for the local fluxes
+ call allocLocal(averageFlux_meta(:)%var_info,flux_mean,nSnow,nSoil,err,cmessage)
+ if(err/=0)then; err=20; message=trim(message)//trim(cmessage); return; endif
```

Comment:
This does eventually backtrack to `multi_driver` and use the `handle_err` error handler.